File: src/types/file.ts
export interface FileNode {
  path: string;
  name: string;
  type: 'file' | 'directory';
  children?: FileNode[];
  size?: number;
  language?: string;
  content?: string;
}

export interface FileItem {
  path: string;
  name: string;
  content: string;
  language: string;
  size: number;
}

export interface FileContent {
  path: string;
  name?: string;
  content: string;
  language?: string;
  size?: number;
}

File: backend/app/schemas/file.py
from pydantic import BaseModel


class FileNodeResponse(BaseModel):
    name: str
    path: str
    type: str  # "file" | "directory"
    size: int | None = None
    extension: str | None = None
    children: list["FileNodeResponse"] | None = None


class FileTreeResponse(BaseModel):
    files: list[FileNodeResponse]


class FileContentResponse(BaseModel):
    name: str
    path: str
    content: str
    language: str | None = None
    size: int
